bot/modals/class_modal.py

```diff
@@ -0,0 +1,164 @@
+            return
+        # create a new class channel scaffold to send to the service
+        scaffold = ClassChannelScaffold(class_prefix=major,
+                                        class_name=title,
+                                        class_number=number,
+                                        class_professor=professor)
+        if self._channel:
+            await self._bot.messenger.publish(
+                Events.on_class_insert, inter, scaffold, self._channel, self._role, description
+            )
+        else:
+            await self._bot.messenger.publish(Events.on_class_create, inter, scaffold, description)
+
+    async def _search_similar(self, inter: discord.Interaction, pref: str, num: int, prof: str) -> bool:
+        similar_class = await self._repo.search_class(pref, num, prof)
+        if not similar_class:
+            return False
+        channel = self._bot.guild.get_channel(similar_class.channel_id)
+        if channel and not similar_class.class_archived:
+            if role := self._bot.guild.get_role(similar_class.class_role_id):
+                await inter.user.add_roles(role)
+            embed = discord.Embed(title='📔 Similar Class Found', color=Colors.Purple)
+            embed.description = f'A similar class channel has been found.'
+            if role:
+                embed.description += f'\nThe {role.mention} role has been added to you.'
+            embed.add_field(name='Channel', value=channel.mention)
+            embed.add_field(name='Instructor', value=similar_class.class_professor)
+            embed.add_field(name='Class Title', value=similar_class.full_title(), inline=False)
+            await inter.response.send_message(embed=embed)
+            return True
+        if channel and similar_class.class_archived:
+            await self._bot.messenger.publish(Events.on_class_unarchive, inter, similar_class)
+            return True
+        return False
+
+    def _autofill(self, data: Union[tuple[str, int], discord.TextChannel]) -> None:
+        if isinstance(data, tuple):
+            prefix, num = data
+            self.major.default = prefix
+            self.course_number.default = num
+            return
+        split_topic = data.topic.split('-') if data.topic else []
+        split_name = data.name.split('-')
+        if len(split_name) >= 3:
+            if valid_course_maj(split_name[0]):
+                self.major.default = split_name[0].upper()
+            if valid_course_num(split_name[1]):
+                self.course_number.default = split_name[1].strip()
+            self.professor.default = split_name[-1].strip().capitalize()[:MAX_INSTR_LEN]
+        if len(split_topic) == 2:
+            self.course_title.default = split_topic[0].title().strip()[:MAX_TITLE_LEN]
+            description = split_topic[1].strip()[:MAX_DESCR_LEN]
+            self.course_description.default = description if len(description) != 0 else 'None'
```

Comment:
```suggestion
            self.course_description.default = description or 'None'
```

Reply:
Oooh, you learn something new every day. Thanks.